From df869a9897a0e935c9288d167f7cab0872f5255d Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Sat, 27 Sep 2025 20:17:36 +0300
Subject: [PATCH 6/6] platform [marvell-prestera][nokia] debian/rules and
 7215-a1 for trixie

NOKIA board 7215-a1
Adjust debian/rules and Kernel-module source for TRIXIE

Signed-off-by: Yan Markman <ymarkman@marvell.com>
---
 .../modules/cn9130_cpu_thermal_sensor.c       |  6 +-
 .../7215-a1/modules/nokia_7215_ixs_a1_cpld.c  |  5 ++
 .../sonic-platform-nokia/debian/rules         | 74 +++++++++----------
 3 files changed, 45 insertions(+), 40 deletions(-)

diff --git a/platform/marvell-prestera/sonic-platform-nokia/7215-a1/modules/cn9130_cpu_thermal_sensor.c b/platform/marvell-prestera/sonic-platform-nokia/7215-a1/modules/cn9130_cpu_thermal_sensor.c
index e11e28157..cbc2fe486 100644
--- a/platform/marvell-prestera/sonic-platform-nokia/7215-a1/modules/cn9130_cpu_thermal_sensor.c
+++ b/platform/marvell-prestera/sonic-platform-nokia/7215-a1/modules/cn9130_cpu_thermal_sensor.c
@@ -57,7 +57,8 @@ static long cn9130_thermal_read_reg_in_mcelcius(struct device *dev, struct cn913
 
     //Read thermal value
     status_regval = readl(temp_base+CN9130_TSEN_REG_STATUS_OFFSET);
-    dev_dbg(dev, "%s: cn9130_thermal_read_reg_in_mcelcius: addr: 0x%lx value:0x%x\n", dev_name(data->hwmon_dev), temp_base+CN9130_TSEN_REG_STATUS_OFFSET, status_regval);
+    dev_dbg(dev, "%s: cn9130_thermal_read_reg_in_mcelcius: addr: %p value:0x%x\n",
+	    dev_name(data->hwmon_dev), (void*)(temp_base+CN9130_TSEN_REG_STATUS_OFFSET), status_regval);
 
     //START MEASUREMENT
     regval = readl(temp_base+CN9130_TSEN_REG_CTRL_0_OFFSET);
@@ -224,7 +225,8 @@ static int __init cn9130_thermal_init_driver(void)
     regval |= 1 << 0; //TSEN_START
     writel(regval, thermal_data->temp_base+CN9130_TSEN_REG_CTRL_0_OFFSET);
 
-    dev_info(dev, "%s: initialized. base_addr: 0x%lx virt_addr:0x%lx\n", dev_name(thermal_data->hwmon_dev), thermal_base_addr, thermal_data->temp_base);
+    dev_info(dev, "%s: initialized. base_addr: 0x%lx virt_addr:%p\n",
+	     dev_name(thermal_data->hwmon_dev), thermal_base_addr, (void *)thermal_data->temp_base);
 
     return 0;
 }
diff --git a/platform/marvell-prestera/sonic-platform-nokia/7215-a1/modules/nokia_7215_ixs_a1_cpld.c b/platform/marvell-prestera/sonic-platform-nokia/7215-a1/modules/nokia_7215_ixs_a1_cpld.c
index e90715dc3..828cfeff7 100644
--- a/platform/marvell-prestera/sonic-platform-nokia/7215-a1/modules/nokia_7215_ixs_a1_cpld.c
+++ b/platform/marvell-prestera/sonic-platform-nokia/7215-a1/modules/nokia_7215_ixs_a1_cpld.c
@@ -42,6 +42,7 @@
 #include <linux/of_device.h>
 #include <linux/of.h>
 #include <linux/mutex.h>
+#include <linux/version.h>
 
 #define DRIVER_NAME "nokia_7215_a1_cpld"
 
@@ -521,8 +522,12 @@ static const struct attribute_group nokia_7215_ixs_a1_cpld_group = {
 };
 
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6,12,0))
+static int nokia_7215_ixs_a1_cpld_probe(struct i2c_client *client)
+#else
 static int nokia_7215_ixs_a1_cpld_probe(struct i2c_client *client,
         const struct i2c_device_id *dev_id)
+#endif
 {
     int status;
      struct cpld_data *data=NULL;
diff --git a/platform/marvell-prestera/sonic-platform-nokia/debian/rules b/platform/marvell-prestera/sonic-platform-nokia/debian/rules
index f3fe01d1b..7dde1c706 100755
--- a/platform/marvell-prestera/sonic-platform-nokia/debian/rules
+++ b/platform/marvell-prestera/sonic-platform-nokia/debian/rules
@@ -22,25 +22,35 @@ PRESTERA_MODULE_PREFIX_ARMHF := mrvl-prestera/drivers/armhf/
 PRESTERA_MODULE_SRC := /cpssEnabler/linuxNoKernelModule/drivers/
 SERVICE_DIR := service
 PLATFORM_DIR := sonic_platform
+#CONF_DIR := conf   - not used
+#UDEV_DIR := udev   - not used
 
 %:
-	dh $@ --with systemd,python3 --buildsystem=pybuild
+	dh $@ --with python3 --buildsystem=pybuild
 
-clean:
-	dh_testdir
-	dh_testroot
-	dh_clean
+override_dh_auto_clean:
 	(for mod in $(MODULE_DIRS); do \
+		$(MAKE) clean -C $(KERNEL_SRC)/build M=$(MOD_SRC_DIR)/$${mod}/modules; \
+		if [ -f $(MOD_SRC_DIR)/$${mod}/setup.py ]; then \
+			PYBUILD_NAME=$${mod} pybuild --clean -d $${mod}; \
+		fi; \
 		cd $(MOD_SRC_DIR)/$${mod}; \
-		rm -rf build/; \
 		rm -rf $(MRVL_MODULE_DIR)/; \
 		rm -rf *.egg-info/; \
 		rm -f *.whl; \
 		cd $(MOD_SRC_DIR)/; \
 	done)
+	dh_clean
 
-build:
+override_dh_auto_configure:
 	(for mod in $(MODULE_DIRS); do \
+		if [ -f $(MOD_SRC_DIR)/$${mod}/setup.py ]; then \
+			PYBUILD_NAME=$${mod} pybuild --configure -d $${mod}; \
+		fi; \
+	done)
+
+override_dh_auto_build:
+	(set -e; for mod in $(MODULE_DIRS); do \
 		mkdir -p $(MOD_SRC_DIR)/$${mod}/$(MRVL_MODULE_DIR); \
 		if [ $$mod = "7215-a1" ] && [ $(CONFIGURED_ARCH) = "arm64" ]; then \
 			cp -r $(MOD_SRC_DIR)/../$(PRESTERA_MODULE_PREFIX)/* $(MOD_SRC_DIR)/$${mod}/$(MRVL_MODULE_DIR)/; \
@@ -61,46 +71,34 @@ build:
 		cd $(MOD_SRC_DIR); \
 	done)
 
-binary: binary-arch binary-indep
-	# Nothing to do
+override_dh_auto_test:
+	# No tests to run
 
-binary-arch:
-	# Nothing to do
+override_dh_usrlocal:
+	# debian/sonic-platform-<>-<>/usr/local/bin/* are already done
 
-binary-indep:
-	dh_testdir
-	dh_installdirs
-
-	# Custom package commands
-	(for mod in $(MODULE_DIRS); do \
+override_dh_auto_install:
+	(set -e; for mod in $(MODULE_DIRS); do \
+		doModInstall=false; \
+		if [ $$mod = "7215-a1" ] && [ $(CONFIGURED_ARCH) = "arm64" ]; then \
+			doModInstall=true; \
+		elif [ $$mod = "7215" ] && [ $(CONFIGURED_ARCH) = "armhf" ]; then \
+			doModInstall=true; \
+		fi; \
+		if [ "$$doModInstall" = false ]; then continue; fi; \
 		dh_installdirs -p$(PACKAGE_PRE_NAME)-$${mod} /$(KERNEL_SRC)/$(INSTALL_MOD_DIR); \
 		dh_installdirs -p$(PACKAGE_PRE_NAME)-$${mod} /usr/local/bin; \
 		dh_installdirs -p$(PACKAGE_PRE_NAME)-$${mod} /boot; \
 		dh_installdirs -p$(PACKAGE_PRE_NAME)-$${mod} /lib/systemd/system; \
 		cp $(MOD_SRC_DIR)/$${mod}/$(SERVICE_DIR)/*.service debian/$(PACKAGE_PRE_NAME)-$${mod}/lib/systemd/system/; \
-		cp $(MOD_SRC_DIR)/$${mod}/$(UTILS_DIR)/* debian/$(PACKAGE_PRE_NAME)-$${mod}/usr/local/bin/; \
+		if [ -d $(MOD_SRC_DIR)/$${mod}/$(UTILS_DIR) ]; then \
+			cp $(MOD_SRC_DIR)/$${mod}/$(UTILS_DIR)/* debian/$(PACKAGE_PRE_NAME)-$${mod}/usr/local/bin/; \
+		fi; \
 		cp $(MOD_SRC_DIR)/$${mod}/$(MRVL_MODULE_DIR)/$(PRESTERA_MODULE_SRC)/mvcpss.ko debian/$(PACKAGE_PRE_NAME)-$${mod}/$(KERNEL_SRC)/$(INSTALL_MOD_DIR); \
-		cp $(MOD_SRC_DIR)/$${mod}/$(MODULE_DIR)/*.ko debian/$(PACKAGE_PRE_NAME)-$${mod}/$(KERNEL_SRC)/$(INSTALL_MOD_DIR); \
+		if [ -d "$(MOD_SRC_DIR)/$${mod}/$(MODULE_DIR)" ] && ls "$(MOD_SRC_DIR)/$${mod}/$(MODULE_DIR)"/*.ko >/dev/null 2>&1; then \
+			cp $(MOD_SRC_DIR)/$${mod}/$(MODULE_DIR)/*.ko debian/$(PACKAGE_PRE_NAME)-$${mod}/$(KERNEL_SRC)/$(INSTALL_MOD_DIR); \
+		fi; \
 		cd $(MOD_SRC_DIR)/$${mod}; \
 		python3 setup.py install --root=$(MOD_SRC_DIR)/debian/$(PACKAGE_PRE_NAME)-$${mod} --install-layout=deb; \
 		cd $(MOD_SRC_DIR); \
 	done)
-
-	# Resuming debhelper scripts
-	dh_testroot
-	dh_install
-	dh_installchangelogs
-	dh_installdocs
-	dh_systemd_enable
-	dh_installinit
-	dh_systemd_start
-	dh_link
-	dh_fixperms
-	dh_compress
-	dh_strip
-	dh_installdeb
-	dh_gencontrol
-	dh_md5sums
-	dh_builddeb
-
-.PHONY: build binary binary-arch binary-indep clean
-- 
2.34.1

